frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn parties_signatures)]
+	pub type PartiesSignatures<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// Vec of judge signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn judge_signatures)]
+	pub(super) type JudgeSignatures<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// List of judges
+	#[pallet::storage]
+	#[pallet::getter(fn judges)]
+	pub type Judges<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxJudges>, ValueQuery>;
+
+	/// The number of contracts that have been made so far.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts_count)]
+	pub type ContractsCount<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, ContractIndex, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Add judge
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `judge`: Account that will be added as one of a judges
+		///
+		/// Emits `AddedJudge`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::add_judge())]
+		pub fn add_judge(origin: OriginFor<T>, judge: T::AccountId) -> DispatchResult {
+			let _ = T::AddJudgeOrigin::ensure_origin(origin)?;
+
+			let real_judges_count = Judges::<T, I>::decode_len().unwrap_or(0) as u32;
+			Judges::<T, I>::try_append(&judge).map_err(|_| Error::<T, I>::TooMany)?;
+
+			Self::deposit_event(Event::AddedJudge {
+				judge_id: real_judges_count + 1,
+				deposit: judge,
+			});
+			Ok(())
+		}
+
+		/// Judge can sign contract
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `contract_id`: The `id` of a contract that should be signed
+		///
+		/// Emits `JudgeSigned`.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::judge_sign_contract())]
+		pub fn judge_sign_contract(
+			origin: OriginFor<T>,
+			contract_id: ContractIndex,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin)?;
+			let is_judge = Judges::<T, I>::get().contains(&who);
+			ensure!(is_judge, Error::<T, I>::NotJudge);
+
+			JudgeSignatures::<T, I>::try_mutate(contract_id, |signatures_list: &mut Option<BoundedVec<T::AccountId, T::MaxSignatures>>| -> DispatchResult {
+				let signatures = signatures_list.as_mut().ok_or(Error::<T, I>::ContractNotFound)?;
+				let already_signed = signatures.contains(&who);
+				ensure!(!already_signed, Error::<T, I>::AlreadySigned);
+
+				signatures.try_push(who.clone()).map_err(|_| Error::<T, I>::TooMany)?;
+				Ok(())
```

Comment:
https://github.com/liberland/liberland_substrate/pull/343#discussion_r1476420779
Still relevant and unaddressed.